backend2/src/api/api_v1/endpoints/articles.py

```diff
@@ -19,10 +24,75 @@ def article_search(
     """
     mutation = item_in.mutation.upper()
     query = db.query(models.ArticleData).filter(models.ArticleData.mutations.any(mutation=mutation))
-    items = [schemas.ArticleSearchItem(
+    items = [article_data_to_search_item(item) for item in query.all()]
+
+    return {mutation: items} if items else {}
+
+
+@router.post("/search-by-file", response_model=schemas.ArticleSearchOut)
+def article_file_search(
+    *,
+    use_snp_effect: bool = Body(False, alias="snpEffect"),
+    file: UploadFile = File(...),
+    db: Session = Depends(deps.get_db),
+) -> schemas.ArticleSearchOut:
+    """
+    Article search by mutations from a VCF-file
+    """
+    parser = VcfParser()
+    try:
+        parser.read_vcf_file(file.file)
+    except AssertionError:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Invalid VCF-file")
+    mutations = parser.get_mutations()
+
+    res = defaultdict(list)
+
+    if use_snp_effect:
```

Comment:
Naming is a bit not clear at this point: protein vs. nucleotide mutations not differentiated.

Reply:
Comments added